server/profile.py

```diff
@@ -442,6 +442,12 @@ def get_ordered_transcript(profile_uc_dict_list):
 
     schedule_screenshot.update_screenshot_async(profile_user)
 
+    scholarships = m.Scholarship.objects()
+    # Filter scholarships based on program
+    scholarships = [s for s in scholarships if profile_user.short_program_name
+            in s.programs and s.id not in profile_user.closed_scholarship_ids]
```

Comment:
nit: Can you add a line before this

```
closed_scholarship_ids_set = set(profile_user.closed_scholarship_ids)
```

... which will hopefully speed things up a tiny bit.